runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/ExecutionStateTracker.java

```diff
@@ -139,8 +140,17 @@ public String getDescription() {
    */
   private volatile long millisSinceLastTransition = 0;
 
+  /**
+   * The number of milliseconds since the {@link ExecutionStateTracker} initial state.
+   *
+   * <p>This variable is updated by the Sampling thread, and read by the Progress Reporting thread,
+   * thus it being marked volatile.
+   */
+  private volatile long millisSinceBundleStart = 0;
```

Comment:
why is this marked as volatile?  And thus needs the supress warning on line 361.

Reply:
Following similar pattern from millisSinceLastTransition

Reply:
its because we want to write from multiple threads during reset.